src/analysis/rule_extraction.jl

```diff
@@ -88,15 +92,18 @@ compact form.
 - `rules` : Vector of Rule objects
 """
 function print_rules(rules::Vector{Rule{Vector{Vector},Vector{Float64}}})::Nothing
-    condition(rule) = [c[2] == :L ? "$(c[1]) < $(c[3])" : "$(c[1]) ≥ $(c[3])" for c in rule.condition]
+    function condition(rule)
```

Comment:
`function` expansion. if you dislike this, I can look into options to prevent it.

Reply:
It's definitely more verbose than I would prefer but I guess it's okay...

In fact, probably a sign that the function should be extracted, as currently it gets recompiled every call to `print_rules()`

Reply:
(that wasn't to say we should refactor it in this PR)